tests/run-tests

```diff
@@ -354,6 +354,7 @@ def run_tests(pyb, tests, args, base_path="."):
 
     if not has_coverage:
         skip_tests.add('cmdline/cmd_parsetree.py')
+        skip_tests.add('unix/schedule_thread.py') # Needs assert()
```

Comment:
This test can still run on any unix variant, right?  Even without assert() enabled in the executable it will still stress test schedule+thread.

And looking at the test itself, seems it just needs threading and scheduler, so could probably be named `tests/thread/stress_schedule.py` to run on any target with threading (eg esp32).

Reply:
Yes, it will run on any build that has _thread and schedule. However, the issue is that the test only fails by triggering the assert. Without the assert the most likely failure mode is an infinite loop as the scheduler stops scheduling. I guess the test could add its own timeout to the main loop?

Reply:
> However, the issue is that the test only fails by triggering the assert. Without the assert the most likely failure mode is an infinite loop as the scheduler stops scheduling.

But at least that's showing a bug.  I think it's important to be able to run this stress test on any port that it can run on.

> I guess the test could add its own timeout to the main loop?

If that's possible, yes.